azerothcore/azerothcore-wotlk
# fix(Core/Unit): Petrified Lichen Guard effect

Petrified Lichen Guard's aura was being applied and procced incorrectly. The aura should no longer overwrite other elemental shields. The aura should now correctly proc its stacking DoT effect.

Closes AzerothCore issue #14415



## Changes Proposed:
-  Resolved false positive in Lightning/Elemental shield conditionals in Unit::HandleProcTriggerSpell() and SpellInfo::LoadSpellSpecific()
-  These were incorrectly causing the Spore Cloud aura from Petrified Lichen Guard to proc the Lightning Shield on-hit effect, and conflict with other elemental auras.

## Issues Addressed:

- Closes #14415
- Closes https://github.com/chromiecraft/chromiecraft/issues/4575

## SOURCE:

https://www.wowhead.com/wotlk/item=25828/petrified-lichen-guard
https://www.wowhead.com/wotlk/spell=32642/spore-cloud

## Tests Performed:

- Built without errors
- Tested on Windows 10 x86_64
- Tested Petrified Lichen Guard on-block effect correctly applied stacking DoT
- Tested Lightning/Earth/Water Shield spells not being overwritten by equipping Petrified Lichen Guard
- Regression tested Lightning/Earth/Water Shield spells still correctly overwriting each other when cast
- Regression tested Lightning/Earth/Water Shield spells still correctly applying their effects

## How to Test the Changes:


1. Create/select a character that can equip shields, and is level 62+
2. .mod rep 970 honored
3. .additem 25828
4. Equip Petrified Lichen Guard and allow mobs to hit you
5. Observe correct application of DoT effect
6. Cast Lightning/Earth/Water Shield spells
7. Observe that the elemental shield spells overwrite each other, but not Petrified Lichen Guard's aura
8. Observe that Lightning/Earth/Water Shield spells still correctly proc their effects

## Known Issues and TODO List:




## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.
